framework/auth/views.py

```diff
@@ -251,6 +288,68 @@ def confirm_email_get(token, auth=None, **kwargs):
     ))
 
 
+@collect_auth
+def unconfirmed_email_remove(auth=None):
+    """Called at login if user cancels their merge or email add.
+    methods: DELETE
+    """
+    user = auth.user
+    if user is None:
+        raise HTTPError(http.NOT_FOUND)
+    json_body = request.get_json()
+    try:
+        given_token = json_body['token']
+    except KeyError:
+        raise HTTPError(http.NOT_FOUND)
```

Comment:
NOT_FOUND (404) is inappropriate for this type of error (the client didn't provide a token)--a 400 should be returned instead. Also, it's good practice to send useful error messages to the client so they know how to correct the error. You can pass `message_short` and `message_long` to the `data` param of `HTTPError`.